lib/main.js

```diff
@@ -189,6 +200,21 @@ export default {
     });
   },
 
+  async getProgram(Linter, configurationPath) {
+    let program;
+    const tsconfigPath = path.resolve(path.dirname(configurationPath), 'tsconfig.json');
+    try {
+      const stats = await stat(tsconfigPath);
+      const isFile = stats.isFile();
+      if (isFile) {
+        program = Linter.createProgram('tsconfig.json', path.dirname(configurationPath));
```

Comment:
You are determining `path.dirname(configurationPath)` twice, just store it in a `const`.